compiler/wasm/src/compile.rs

```diff
@@ -24,14 +28,26 @@ extern "C" {
     pub type StringArray;
 }
 
+#[derive(Deserialize)]
+struct DependencyGraph {
+    root_dependencies: Vec<String>,
+    library_dependencies: HashMap<String, Vec<String>>,
+}
+
 #[wasm_bindgen]
 pub fn compile(
     entry_point: String,
     contracts: Option<bool>,
-    dependencies: Option<StringArray>,
+    dependency_graph: JsValue,
```

Comment:
This will be typed as `any` in JS which means that we're giving up type safety on the dependency graph. Can you define a proper type for this?

Reply:
Ideally this would be something like `StringArray` of `JsWitnessMap` where we have explicit logic on how to convert from the JS to Rust representations. It would also be acceptable to have something like `JsAbi` where we just define an explicit TS type and use serde.